### PR TITLE
Check in ImageContentSourcePolicy for nightly installations

### PR DESCRIPTION
**Pull request by pierDipi**
I need to apply this to install the nightly index with:
```yaml
apiVersion: v1
kind: Namespace
metadata:
  name: openshift-serverless
---
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: serverless-operator
  namespace: openshift-marketplace
spec:
  displayName: Serverless Operator
  image: registry.ci.openshift.org/knative/main:serverless-index
  publisher: Red Hat
  sourceType: grpc
---
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: serverless
  namespace: openshift-serverless
---
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: serverless-operator
  namespace: openshift-serverless
spec:
  channel: "stable"
  name: "serverless-operator"
  source: "serverless-operator"
  sourceNamespace: "openshift-marketplace"
  installPlanApproval: Automatic
```

### CODE REVIEW

File: hack/generate/catalog.sh [hunk 1]
```diff
@@ -155,4 +155,8 @@ EOF
   fi
 }
 
+logger.info "Generating ImageContextSourcePolicy"
+create_image_content_source_policy "registry.ci.openshift.org/knative/${CURRENT_VERSION_IMAGES}:serverless-index" "$registry_redhat_io" "$registry_quay" "olm-catalog/serverless-operator-index/image_content_source_policy.yaml"
```
Comment (mgencur) [inline hack/generate/catalog.sh:159]: This file is not applied anywhere. Will this be applied in some other repository?

Comment (pierDipi) [reply hack/generate/catalog.sh:159]: Yes, it's mostly for development and demos like https://github.com/keventmesh/eventing-platform-architecture

**Pull request merged**